benchmarking-tests/benchmark_utils.py

```diff
@@ -36,6 +44,57 @@ def run_set_statsd_host(args):
         json.dump(conf, f)
 
 
+def run_gather_metrics(args):
+    # setup a rethinkdb connection
+    conn = r.connect(args.bigchaindb_host, 28015, 'bigchain')
+
+    # setup csv writer
+    csv_file = open(args.csvfile, 'w')
+    csv_writer = csv.writer(csv_file)
+
+    # query for the number of transactions on the backlog
+    num_transactions = r.table('backlog').count().run(conn)
+    num_transactions_received = 0
+    initial_time = None
+    logger.info('Starting gathering metrics. {} transasctions in the backlog'.format(num_transactions))
+    logger.info('This process should exit automatically. '
+                'If this does not happen you can exit at any time using Ctrl-C'
+                ' saving all the metrics gathered up to this point.')
+
+    logger.info('\t{:<20} {:<20} {:<20} {:<20}'.format('timestamp', 'tx in block',
+                                                     'tx/s', '% complete'))
+
+    # listen to the changefeed
+    try:
+        for change in r.table('bigchain').changes().run(conn):
+            # check only for new blocks
+            if change['old_val'] is None:
+                block_num_transactions = len(change['new_val']['block']['transactions'])
+                time_now = time.time()
+                csv_writer.writerow([str(time_now), str(block_num_transactions)])
+
+                # log statistics
+                if initial_time is None:
+                    initial_time = time_now
+
+                num_transactions_received += block_num_transactions
+                elapsed_time = time_now - initial_time
+                elapsed_time = elapsed_time if elapsed_time != 0 else 1
+                percent_complete = round((num_transactions_received / num_transactions) * 100)
+                transactions_per_second = round(num_transactions_received / elapsed_time)
+
+                logger.info('\t{:<20} {:<20} {:<20} {:<20}'.format(time_now, block_num_transactions,
+                                                                   transactions_per_second, percent_complete))
+
+                if (num_transactions - num_transactions_received) == 0:
+                    break
+    except KeyboardInterrupt:
+        logger.info('Interrupted. Exiting early...')
+
+    # close files
+    csv_file.close()
```

Comment:
Maybe the `csv_file.close()` should go inside [a `finally` clause](https://docs.python.org/3/tutorial/errors.html#defining-clean-up-actions) (clean-up actions that must be executed under all circumstances), i.e.
```python
    finally:
        csv_file.close()
```